ffi/freetype.lua

```diff
@@ -186,18 +215,71 @@ end
 
 local FTFaceType = ffi.metatype("struct FT_FaceRec_", FTFace_mt) -- luacheck: ignore 211
 
+local FTSize_mt = {__index={}}
+
+function FTSize_mt.__index:checkGlyph(char)
+    assert(ft2.FT_Activate_Size(self) == 0, "failed to activate font size")
```

Comment:
Sure `FT_Activate_Size()` is really cheap when it is already the active FT_Size, as we'll be calling it often ? And that we don't need to store on the Lua side which is active to avoid calling it ?

Reply:
Yes:
```C
  FT_EXPORT_DEF( FT_Error )
  FT_Activate_Size( FT_Size  size )
  {
    FT_Face  face;

    if ( !size )
      return FT_THROW( Invalid_Size_Handle );

    face = size->face;
    if ( !face || !face->driver )
      return FT_THROW( Invalid_Face_Handle );

    /* we don't need anything more complex than that; all size objects */
    /* are already listed by the face                                  */
    face->size = size;

    return FT_Err_Ok;
  }
```

The alternative would be to not do it automatically, and add an explicit call to `ftsize:activate()` before using any other methods.